### DISCUSSION TITLE
Preview build from main branch

### DISCUSSION THREAD

**Discussion by dani-mp** [vercel/next.js#31468] [Help]
Hey! I can't figure this out so I'm sure I'm doing something wrong.

My repo's default branch is `main`, and when I push changes to it, a production build is deployed to Vercel.

Apart from this, when I push changes to `main` I also want to deploy a preview build to a different domain, with different env vars. This seems to be impossible, right?

I could automatically mirror my `main` branch with another branch and set up a preview deployment from this one, but that's not ideal because I don't want to keep two branches around with the same code, it doesn't make sense. What am I missing?
 
Thanks a lot.

**Comment by AndreMiras** [comment]
Yeah I also had that need, but I'm not sure Vercel is designed for that.
Another way to make it work is to create a second Vercel project also pointing to main, but with the preview environment variables.
However you would also need to "disable" preview deployments on one of the two projects or you would have two previews deployments when pull requesting and branching

**Comment by michaelknoch** [comment]
Did you find a workaround? Feels quite weird to not have this feature. How would you implement CMS preview without it?

**Comment by DavidDeSloovere** [comment]
There are GH actions that can do this, i.e. https://github.com/marketplace/actions/sync-branch
